### PR TITLE
Migrate Jenkins jobs to GitHub Actions

### PR DESCRIPTION
**Pull request by andreydevyatkin**
The following jenkins jobs have been migrated:
- https://ci-beam.apache.org/job/beam_PerformanceTests_Cdap/
- https://ci-beam.apache.org/job/beam_PerformanceTests_HadoopFormat/
- https://ci-beam.apache.org/job/beam_PerformanceTests_MongoDBIO_IT/
- https://ci-beam.apache.org/job/beam_PerformanceTests_PubsubIOIT_Python_Streaming/
- https://ci-beam.apache.org/job/beam_PerformanceTests_SparkReceiver_IO/
- https://ci-beam.apache.org/job/beam_PerformanceTests_WordCountIT_Py38/
- beam_PerformanceTests_TFRecordIOIT_HDFS (removed in Jenkins)

### How has it been tested?

The jobs have been tested on Self-hosted GKE runners ubuntu 20.04.
The following test cases are passed:

- by manually running the job (`workflow_dispatch`)

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: .github/workflows/beam_PerformanceTests_Cdap.yml
Comment (Abacn) [inline .github/workflows/beam_PerformanceTests_Cdap.yml:108]: There are a couple of leftover workloads likely due to manual run. 

image

and I do not see a teardown step in these workflows. Please check

Comment (Abacn) [reply .github/workflows/beam_PerformanceTests_Cdap.yml:108]: note that some are due to https://github.com/apache/beam/blob/master/.github/workflows/beam_PerformanceTests_AvroIOIT_HDFS.yml on master, not this PR though

Comment (volatilemolotov) [reply .github/workflows/beam_PerformanceTests_Cdap.yml:108]: @Abacn Teardown is in the setup-k8s-access action and needs approval for

pytooling/actions/with-post-step@v0.4.6

Comment (andreydevyatkin) [reply .github/workflows/beam_PerformanceTests_Cdap.yml:108]: @Abacn @volatilemolotov These namespaces are from both Jenkins and GitHub Actions - the namespaces which have a short build numbers (e.g. `beam-python-mongoio-load-test-5071`) are Jenkins ones and the long build numbers are related to GH Actions (e.g. `beam-performancetests-avroioit-hdfs-6303887955`). The GH Actions namespaces are listed because the `Post Set k8s access` step was not triggered due to the cancelled run in our forks:
- https://github.com/andreydevyatkin/beam/actions/runs/6303887955
- https://github.com/andreydevyatkin/beam/actions/runs/6324129155

At this moment, we are waiting for approval from the infra team to whitelist the `setup-k8s-access` action https://github.com/apache/beam/pull/28530#issuecomment-1734560209

Comment (Abacn) [reply .github/workflows/beam_PerformanceTests_Cdap.yml:108]: Thanks for clarification

**Review by Abacn — APPROVED** [review]
pytool action is enabled and saw a green run for Avro_HDFS prformance tests, so merge them also

**Pull request merged**
